From fe33995a9af93beb206a65ef25ffcc351f80d10e Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Wed, 27 Aug 2025 16:56:40 +0200
Subject: [PATCH 041/553] util/unicode: readEastAsianWidth(): remove
 simplified() call

All users of the split()ed value handle intervening whitespace
already:

- fields[0] is piped through parseHexRange(), which does

- fields[1] has trimmed() called on it before lookup and all
  idnaStatusMap values are space-free (cf. initIdnaStatusMap())

As a consequence, we can accept the line by reference to const
QByteArray now.

Amends 838a7a01f388673c45af2dd60bc38992de1a2a05.

Pick-to: 6.9 6.8 6.5
Change-Id: I53247332c624a192fcaca6009a3f20cb8c65786a
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit 2c3c479c202731b7ea9446945706769f0a04161c)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 util/unicode/main.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/util/unicode/main.cpp b/util/unicode/main.cpp
index c8c3383702b..73c39e6554e 100644
--- a/util/unicode/main.cpp
+++ b/util/unicode/main.cpp
@@ -1629,9 +1629,7 @@ static void readDerivedAge()
 static void readEastAsianWidth()
 {
     readUnicodeFile("EastAsianWidth.txt",
-                    [] (QByteArray &line, int lineNo) {
-        line = std::move(line).simplified();
-
+                    [] (const QByteArray &line, int lineNo) {
         QList<QByteArray> fields = line.split(';');
         Q_ASSERT(fields.size() == 2);
 
-- 
2.51.2

